web/modules/custom/search_api_field/src/Plugin/facets/facet_source/SearchApiFieldDeriver.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\search_api_field\Plugin\facets\facet_source;
+
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\facets\FacetSource\FacetSourceDeriverBase;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Derives a facet source plugin definition for every search api page.
+ *
+ * The definition of this plugin happens in facet_source\SearchApiPage, in this
+ * deriver class we're actually getting all possible pages and creating plugins
+ * for each of them.
+ *
+ * @see \Drupal\search_api_page\Plugin\facets\facet_source\SearchApiPage
+ */
+class SearchApiFieldDeriver extends FacetSourceDeriverBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $base_plugin_id = $base_plugin_definition['id'];
+
+    if (!isset($this->derivatives[$base_plugin_id])) {
+
+      $map = \Drupal::entityManager()->getFieldMapByFieldType('search');
```

Comment:
Please use dependency injection, and don't use the deprecated `EntityManager` service, but rather `EntityFieldManager`.